### PR TITLE
CompactFormatter support for LogRecord::getLongThreadID #529

### PR DESCRIPTION
**Pull request by jmehrens**
This is the 1.x backport of https://github.com/eclipse-ee4j/mail/issues/529

### CODE REVIEW

**Comment by jmehrens** [comment]
@lukasj This can be merged into 1.x.  There are lots of commits so it needs to be squashed and merged to get rid of them.

**Comment by jmehrens** [comment]
I was able to test under JDK7 and JDK8 all tests pass related to this PR.  https://github.com/eclipse-ee4j/mail/issues/548 is failing only under JDK7 and it is unrelated to these changes.

File: mail/src/main/java/com/sun/mail/util/logging/LogManagerProperties.java [hunk 1]
```diff
@@ -17,6 +17,10 @@
 package com.sun.mail.util.logging;
 
 import java.io.*;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.invoke.MethodType;
```
Comment (lukasj) [inline mail/src/main/java/com/sun/mail/util/logging/LogManagerProperties.java:23]: @jmehrens this is causing android sigtests to fail. Do you want me to rollback this PR from 1.6.x or do you prefer to fix it?

Comment (jmehrens) [reply mail/src/main/java/com/sun/mail/util/logging/LogManagerProperties.java:23]: @lukasj Looks like the Android API level 26 is where MethodHandles were added.  I'll update to use reflection for 1.x.  I should be able to get it patched tonight.

**Pull request merged**